sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -324,6 +325,15 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
     }
   }
 
+  // Try to output nvidia-smi output when task fails due to a cuda exception.
+  private def logGpuDebugInfo() = {
+    try {
+      val nvidiaSmiOutput = "nvidia-smi".!!
```

Comment:
I think it does not capture stderr.

When there is an error I also assume there is a non-negligible chance that nvidia-smi blocks for a long time or indefinitely. Do we need a time out?

Reply:
That's a good idea.  We don't want to hang in the call back.